unity/Assets/Scripts/MCSSimulationAgent.cs

```
@@ -34,14 +34,34 @@ public class MCSSimulationAgent : MonoBehaviour {
     private SkinObjectMaterialOption legs = null;
     private int skin = 0;
     private ObjectMaterialOption tie = null;
-    private Animator animator;
+    private MCSMain mcsMain;
+
 
-    private int currentAnimationFrame = 0;
-    private static int ANIMATION_FRAME_RATE = 25;
+    private Animator animator;
+    public static int ANIMATION_FRAME_RATE = 25;
+    public static int AGENT_INTERACTION_ACTION_STARTING_ANIMATION_FRAME = 3;
+    public int currentAnimationFrame = 0;
     [SerializeField] private string currentClip;
-    private Dictionary<string, float> clipNamesAndDurations = new Dictionary<string,float>();
+    public Dictionary<string, float> clipNamesAndDurations = new Dictionary<string,float>();
     private bool resetAnimationToIdleAfterPlayingOnce = false;
     private int stepToEndAnimation = -1;
+    
+    
+    //local position adjustments of the held object throughout the animation sequence, calculated by hand in the editor
+    private static Vector3 REACH_INTO_BACK_POSITION_1 = new Vector3(0.0057f, 1.41f, -0.156f);
+    private static Vector3 REACH_INTO_BACK_POSITION_2 = new Vector3(0.1296f, 1.4736f, 0.1247f);
+    private static Vector3 REACH_INTO_BACK_POSITION_3 = new Vector3(0.032f, 1.417f, 0.191f);
+    private static Vector3 HOLDING_POSITION = new Vector3(0.002f, 1.119f, 0.1907f);
```

Have we tested any of this out on objects that aren't soccer balls or don't have scale (1, 1, 1)? I'm assuming these values might not work for any object.

We don't currently have a need for non-soccer-balls or large soccer balls, so it's perfectly fine with me if we want to hard-code a restriction on the object's type and/or bounding box scale (maybe throw an error in MCSMain if `objectConfig.associatedWithAgent != null` but the object is restricted?). However, we should make sure to update the docs for `associatedWithAgent` about the restrictions if we add any.

---

I have tested it and it looks a little clunky with the object floating at certain points or clearly clipping into the agents head. These are with blocks or the turtle. I think restricting to balls is the best idea for now with a bounding box scale between 0.2 and 0.25. I will update the docs and throw an error.